messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -255,4 +292,60 @@ protected void createCheckpoint() {
       currentCheckpoint = new CheckpointList(partitionMinList);
     }
   }
+
+  @Override
+  public synchronized void close() {
+    newReaders.clear();
+    if (null != partitionDiscovererAndReaderCreator) {
+      partitionDiscovererAndReaderCreator.cancel();
+    }
+    super.close();
+  }
+
+  protected void startNewReaders() throws IOException{
+    for (PartitionId id : newReaders.keySet()) {
+      PartitionReader reader = newReaders.get(id);
+      reader.start(getReaderNameSuffix());
+      readers.put(id, reader);
+      LOG.info("started new reader " + getReaderNameSuffix() +
+          " " + reader.toString());
```

Comment:
But I would not prefer a transient runtime numeric value (hashcode) representing a PartitionReader instance in logs. I would instead prefer using PartitionId.toString(). We can rephrase the log to:
"started new reader " + getReaderNameSuffix() + for partition " + id.toString();